symengine/series_generic.cpp

```diff
@@ -1,166 +1,197 @@
+    Expression i2(x.get_den());
+    i1 /= i2;
+    return i1;
 }
 
-RCP<const UnivariateSeries> neg_uni_series (const UnivariateSeries& a)
-{
-    return make_rcp<const UnivariateSeries>(a.var_, a.prec_, std::move(neg_uni_poly(*a.poly_)));
+Expression UnivariateSeries::convert(const Number &x) {
+    //Expression i1(x));
+    //Expression i2(x.get_den());
+    //i1 /= i2;
+    //return Expression(x);
+    throw std::runtime_error("Not Implemented");
 }
 
-RCP<const UnivariateSeries> sub_uni_series (const UnivariateSeries& a, const UnivariateSeries& b)
-{
-    return add_uni_series(a, *neg_uni_series(b));
+map_uint_Expr UnivariateSeries::convert_map(const map_uint_mpz &d) {
+    map_uint_Expr dict;
+    for (const auto &it : d)
+        dict[it.first] = Expression(it.second.get_ui());
+    return dict;
 }
 
-RCP<const UnivariateSeries> mul_uni_series (const UnivariateSeries& a, const UnivariateSeries& b)
-{
-    map_uint_mpz dict;
-    SYMENGINE_ASSERT(a.var_->get_name() == b.var_->get_name())
-    const unsigned int minprec = (a.prec_ < b.prec_)? a.prec_ : b.prec_;
+RCP<const UnivariatePolynomial> UnivariateSeries::convert_poly(const map_uint_mpz &d) {
+    map_uint_Expr dict_trunc;
     unsigned int max = 0;
-    for (const auto &ait : a.poly_->dict_) {
-        const unsigned int aexp = ait.first;
-        if (aexp < minprec) {
-            for (const auto &bit : b.poly_->dict_) {
-                const unsigned int expsum = aexp + bit.first;
-                if (expsum < minprec)
-                    mpz_addmul(dict[expsum].get_mpz_t(), ait.second.get_mpz_t(), bit.second.get_mpz_t());
-                else
-                    break;
-                if (expsum > max)
-                     max = expsum;
-            }
+    for (const auto &it : d) {
+        if (it.first < prec_) {
+            if (max < it.first)
+                max = it.first;
+            dict_trunc[it.first] = it.second.get_ui();
         }
-        else
-            break;
     }
-    return make_rcp<const UnivariateSeries>(a.var_, minprec, max, std::move(dict));
+    return univariate_polynomial(symbol(var_), max, std::move(dict_trunc));
+}
+
+RCP<const UnivariatePolynomial> UnivariateSeries::convert_vector(const std::vector<mpz_class> &v) {
+    std::vector<Expression> vtrunc;
+    for (const auto &it : v)
+        if (it.get_si() < prec_) 
+            vtrunc.push_back(it.get_si());
+    return UnivariatePolynomial::create(symbol(var_), vtrunc);
+}
+
+unsigned UnivariateSeries::ldegree(const UnivariateExprPolynomial &s) {
+    return s.get_univariate_poly()->get_dict().begin()->first;
+}
+
+UnivariateExprPolynomial UnivariateSeries::mul(const UnivariateExprPolynomial &s, const UnivariateExprPolynomial &r, unsigned prec) {
+    // No prec mul
+    return s * r;
+}
+
+UnivariateExprPolynomial UnivariateSeries::pow(const UnivariateExprPolynomial &s, int n, unsigned prec) {
+    // No prec mul
+    return pow_ex(Expression(s.get_basic()), Expression(n));
```

Comment:
`pow` is for exponentiation of polynomials. So, you need to implement `pow_poly`